src/gc.c

```diff
@@ -833,6 +833,7 @@ void jl_gc_track_malloced_array(jl_ptls_t ptls, jl_array_t *a)
         ptls->heap.mafreelist = ma->next;
     }
     ma->a = a;
+    ma->how = how;
```

Comment:
Both `a` and `next` are aligned (4 bits in `a` and 2 bits in `next`). You can pick your favorate one ;-p